cmd/jaeger/internal/integration/trace_writer_test.go

```diff
@@ -0,0 +1,125 @@
+// Copyright (c) 2024 The Jaeger Authors.
+// SPDX-License-Identifier: Apache-2.0
+
+package integration
+
+import (
+	"context"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/mock"
+	"github.com/stretchr/testify/require"
+	"go.opentelemetry.io/collector/component"
+	"go.opentelemetry.io/collector/consumer"
+	"go.opentelemetry.io/collector/pdata/pcommon"
+	"go.opentelemetry.io/collector/pdata/ptrace"
+	"go.uber.org/zap/zaptest"
+)
+
+type MockExporter struct {
+	mock.Mock
+	chunks []ptrace.Traces
+}
+
+func (m *MockExporter) ConsumeTraces(ctx context.Context, td ptrace.Traces) error {
+	m.chunks = append(m.chunks, td)
+	args := m.Called(ctx, td)
+	return args.Error(0)
+}
+
+func (m *MockExporter) Capabilities() consumer.Capabilities {
+	args := m.Called()
+	return args.Get(0).(consumer.Capabilities)
+}
+
+func (*MockExporter) Shutdown(context.Context) error {
+	return nil
+}
+
+func (*MockExporter) Start(context.Context, component.Host) error {
+	return nil
+}
+
+func TestWriteTracesInChunks(t *testing.T) {
+	td := ptrace.NewTraces()
+	rspans := td.ResourceSpans()
+
+	rspan1 := rspans.AppendEmpty()
+	rspan1.Resource().Attributes().PutStr("service.name", "NoServiceName1")
+	scope1 := rspan1.ScopeSpans()
+	sspan1 := scope1.AppendEmpty()
+	sspan1.Scope().SetName("success-op-1")
+	span1 := sspan1.Spans().AppendEmpty()
+	span1.SetName("span1")
+	span1.SetSpanID(pcommon.SpanID([8]byte{0, 0, 0, 0, 0, 0, 0, 1}))
+	span2 := sspan1.Spans().AppendEmpty()
+	span2.SetName("span2")
+	span2.SetSpanID(pcommon.SpanID([8]byte{0, 0, 0, 0, 0, 0, 0, 2}))
+
+	rspan2 := rspans.AppendEmpty()
+	rspan2.Resource().Attributes().PutStr("service.name", "NoServiceName2")
+	scope2 := rspan2.ScopeSpans()
+	sspan2 := scope2.AppendEmpty()
+	sspan2.Scope().SetName("success-op-2")
+	span3 := sspan2.Spans().AppendEmpty()
+	span3.SetName("span3")
+	span3.SetSpanID(pcommon.SpanID([8]byte{0, 0, 0, 0, 0, 0, 0, 3}))
+
+	rspan3 := rspans.AppendEmpty()
+	rspan3.Resource().Attributes().PutStr("service.name", "NoServiceName3")
+	scope3 := rspan3.ScopeSpans()
+	sspan3 := scope3.AppendEmpty()
+	sspan3.Scope().SetName("success-op-3")
+	span4 := sspan3.Spans().AppendEmpty()
+	span4.SetName("span4")
+	span4.SetSpanID(pcommon.SpanID([8]byte{0, 0, 0, 0, 0, 0, 0, 4}))
```

Comment:
up to this point it looks like you're creating 3 services with two spans each. I would do it in a loop that makes it much more obvious.